### PR TITLE
Or 1807 fix issues from the internal audit s feedback

### PR DESCRIPTION
**Pull request by nguyenzung**
Apply updated code to the codeReview branch

Thank you!

### CODE REVIEW

File: packages/tokamak/contracts-bedrock/src/L1/OptimismPortal.sol [hunk 1]
```diff
@@ -355,8 +403,7 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
         // Set the l2Sender so contracts know who triggered this withdrawal on L2.
         l2Sender = _tx.sender;
         if (_tx.value > 0) {
-            IERC20(_nativeTokenAddress).safeIncreaseAllowance(_tx.target, _tx.value);
-            depositedAmount -= _tx.value;
+            IERC20(_nativeTokenAddress).safeIn(_tx.target, _tx.value);
```
Comment (rlgns98kr) [inline packages/tokamak/contracts-bedrock/src/L1/OptimismPortal.sol:406]: I think we can fix it! 🙇🏻‍♂️🙇🏻‍♂️

Comment (nguyenzung) [reply packages/tokamak/contracts-bedrock/src/L1/OptimismPortal.sol]: Thank you very much, @rlgns98kr 
I updated!

**Comment by nguyenzung** [comment]
Hi @theo-learner 
The base branch is read-only. Could you update the permission?

**Comment by theo-learner** [comment]
> Hi @theo-learner The base branch is read-only. Could you update the permission?

Ok. Let me check and update it

**Comment by theo-learner** [comment]
@nguyenzung It it OK to change target branch from codeReview to main? We have to keep the branch permission of codeReview as project FBI's policy.

**Comment by nguyenzung** [comment]
> @nguyenzung Is it OK to change target branch from codeReview to main? We have to keep the branch permission of codeReview as project FBI's policy.

I think it is okay @theo-learner 

I will make other PR to merge the updated code to the main branch. 

I make this PR so that we can review the result after internal audit easier arcoding @0x6e616d and @rlgns98kr  opinions. 

I will close this PR after every member verifies the result.

**Review by rlgns98kr — APPROVED** [review]
It's great! Thank you very much, Brave!

I think we can merge it to main!

**Comment by rlgns98kr** [comment]
> 훌륭해요! 정말 고맙습니다, Brave!
> 
> 메인에 합칠 수 있을 것 같아요!



> > @nguyenzung Is it OK to change target branch from codeReview to main? We have to keep the branch permission of codeReview as project FBI's policy.
> 
> I think it is okay @theo-learner
> 
> I will make other PR to merge the updated code to the main branch.
> 
> I make this PR so that we can review the result after internal audit easier arcoding @0x6e616d and @rlgns98kr opinions.
> 
> I will close this PR after every member verifies the result.
> 
> Let me think about how to merge to OR-1810. Thank you very much, Theo!

And I agree your opinion. We can close this PR instead of editing base branch!

**Pull request closed without merging**